dotCMS/src/test/java/com/dotmarketing/filters/VanityUrlRequestWrapperTest.java

```diff
@@ -0,0 +1,100 @@
+package com.dotmarketing.filters;
+
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.junit.Test;
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.request.MockParameterRequest;
+import com.dotcms.vanityurl.model.VanityUrlResult;
+import com.google.common.collect.ImmutableMap;
+
+public class VanityUrlRequestWrapperTest {
+
+    final String URL = "URL";
+    final String FORM = "FORM";
+    final String VANITY = "VANITY";
+
+
+    /**
+     * This tests if you have a vanity URL that has query parameters, that they will be merged into to
+     * the request and if their keys match an existing parameter, the vanity url parameter will
+     * overwrite any parameters that were in the original request
+     * 
+     * see: https://github.com/dotCMS/core/issues/18325
+     */
+    @Test
+    public void test_that_vanity_urls_with_query_params_override_incoming_requests() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5391667)